test/integration/isolated/ingress_test.go

```diff
@@ -196,3 +200,155 @@ func TestIngressGRPC(t *testing.T) {
+						},
+					},
+					Spec: incubatorv1alpha1.KongServiceFacadeSpec{
+						Backend: incubatorv1alpha1.KongServiceFacadeBackend{
+							Name: service.Name,
+							Port: test.HTTPBinPort,
+						},
+					},
+				},
+			}
+			clients := GetFromCtxForT[*clientset.Clientset](ctx, t)
+			for _, serviceFacade := range serviceFacades {
+				serviceFacade, err = clients.IncubatorV1alpha1().KongServiceFacades(namespace).Create(ctx, serviceFacade, metav1.CreateOptions{})
+				require.NoError(t, err)
+				cleaner.Add(serviceFacade)
+			}
+
+			t.Log("configuring Ingress")
+			ingress := builder.NewIngress(uuid.NewString(), ingressClass).WithRules(
+				netv1.IngressRule{
+					IngressRuleValue: netv1.IngressRuleValue{
+						HTTP: &netv1.HTTPIngressRuleValue{
+							Paths: []netv1.HTTPIngressPath{
+								{
+									Path:     "/jpeg",
+									PathType: lo.ToPtr(netv1.PathTypePrefix),
+									Backend: netv1.IngressBackend{
+										Resource: &corev1.TypedLocalObjectReference{
+											APIGroup: lo.ToPtr(incubatorv1alpha1.SchemeGroupVersion.Group),
+											Kind:     incubatorv1alpha1.KongServiceFacadeKind,
+											Name:     serviceFacades[0].Name,
+										},
+									},
+								},
+								{
+									Path:     "/png",
+									PathType: lo.ToPtr(netv1.PathTypePrefix),
+									Backend: netv1.IngressBackend{
+										Resource: &corev1.TypedLocalObjectReference{
+											APIGroup: lo.ToPtr(incubatorv1alpha1.SchemeGroupVersion.Group),
+											Kind:     incubatorv1alpha1.KongServiceFacadeKind,
+											Name:     serviceFacades[1].Name,
+										},
+									},
+								},
+							},
+						},
+					},
+				}).
+				WithAnnotations(map[string]string{
+					annotations.AnnotationPrefix + annotations.StripPathKey: "true",
+				}).Build()
+			ingress, err = cluster.Client().NetworkingV1().Ingresses(namespace).Create(ctx, ingress, metav1.CreateOptions{})
+			require.NoError(t, err)
+			cleaner.Add(ingress)
+
+			return ctx
+		}).
+		Assess("KongServiceFacades annotations work", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			proxyURL := GetProxyURLFromCtx(ctx)
+			httpClient := helpers.DefaultHTTPClient()
+
+			expectContentType := func(path, expectedContentType string) {
+				assert.Eventually(t, func() bool {
+					resp, err := httpClient.Get(fmt.Sprintf("%s%s", proxyURL, path))
+					if err != nil {
+						t.Log(err)
+						return false
+					}
+					defer resp.Body.Close()
+
+					if contentType := resp.Header.Get("Content-Type"); contentType != expectedContentType {
+						t.Logf("expected Content-Type header to be %s, got %s", expectedContentType, contentType)
+						return false
+					}
+					return true
+				}, consts.IngressWait, consts.WaitTick)
+			}
+
+			t.Log("asserting /jpeg path returns JPEG image")
+			expectContentType("/jpeg", "image/jpeg")
+
+			t.Log("asserting /png path returns PNG image")
+			expectContentType("/png", "image/png")
```

Comment:
You can write it with an existing helper like e.g. here https://github.com/Kong/kubernetes-ingress-controller/blob/428f171662e977a9d8d970457901ea08b529661a/test/integration/ingress_test.go#L1183
only magic number has to be moved to helper package and made public